ee/session_recordings/session_summary/input_data.py

```diff
@@ -0,0 +1,173 @@
+    if not local_path:
+        session_metadata = SessionReplayEvents().get_metadata(session_id=str(session_id), team=team)
+    else:
+        raw_session_metadata = load_session_metadata_from_json(local_path)
+        session_metadata = cast(RecordingMetadata, raw_session_metadata)
+    if not session_metadata:
+        raise ValueError(f"No session metadata found for session_id {session_id}")
+    return session_metadata
+
+
+def get_session_events(
+    session_id: str, session_metadata: RecordingMetadata, team: Team, local_path: str | None = None
+) -> tuple[list[str], list[tuple[str | datetime | list[str] | None, ...]]]:
+    if not local_path:
+        session_events_columns, session_events = SessionReplayEvents().get_events(
+            session_id=str(session_id),
+            team=team,
+            metadata=session_metadata,
+            events_to_ignore=[
+                "$feature_flag_called",
+            ],
+            extra_fields=EXTRA_SUMMARY_EVENT_FIELDS,
+        )
+    else:
+        session_events_columns, session_events = load_session_recording_events_from_csv(
+            local_path, extra_fields=EXTRA_SUMMARY_EVENT_FIELDS
+        )
+    if not session_events_columns or not session_events:
+        raise ValueError(f"no events found for session_id {session_id}")
+    return session_events_columns, session_events
+
+
+def _skip_event_without_context(
+    event_row: list[str | datetime | list[str] | None],
+    # Using indexes as argument to avoid calling get_column_index on each event row
+    indexes: dict[str, int],
+) -> bool:
+    """
+    Avoid events that don't add meaningful context and confuse the LLM.
+    """
+    event = cast(str, event_row[indexes["event"]])
+    elements_chain_texts = event_row[indexes["elements_chain_texts"]]
+    elements_chain_elements = event_row[indexes["elements_chain_elements"]]
+    elements_chain_href = event_row[indexes["elements_chain_href"]]
+    elements_chain_ids = event_row[indexes["elements_chain_ids"]]
+    # Skip autocapture events with no proper context
+    if event == "$autocapture":
+        if (
+            not elements_chain_texts
+            and not elements_chain_elements
+            and not elements_chain_href
+            and not elements_chain_ids
+        ):
+            return True
+    # Never skip system events
+    if event.startswith("$"):
+        return False
+    # Never skip events with descriptive names
+    if len(event.split(" ")) > 1 or len(event.split(".")) > 1 or len(event.split("_")) > 1:
+        return False
+    # Skip the events with a short name or no context, assuming they aren't useful for the summary
+    # TODO: Check the assumption, as could be risky, but worth it to avoid adding noise
+    if not elements_chain_texts and not elements_chain_elements and not elements_chain_href and not elements_chain_ids:
+        return True
```

Comment:
skip comment here not to match the following code and duplicates line 65 above?

Reply:
Yup, good catch, excessive check, will remove.

Reply:
[`e162f2e` (#31394)](https://github.com/PostHog/posthog/pull/31394/commits/e162f2e330797bb84c7cb3634d84743931477430)